library-best-practices/glossary.md

```diff
@@ -66,3 +66,77 @@ Java Dependency Glossary
 - **Linkage-matchable version** (said of a particular version of A in relation
   to all versions of B): There exists some version of B such that the version of
   A and the version of B are linkage-compatible.
+
+
+### Static Linkage Check
+
+- **Static linkage check**: a process to identify the existence of static
+  linkage errors for a given classpath, by scanning Java classes and
+  verifying the availability in the classpath.
+
+- **Static linkage report**: the outcome of a static linkage check. It reports
+  zero or more static linkage errors.
+
+- **Static linkage errors**: a reported error for a reference as the result of
+  a static linkage check.
+  A static linkage error contains the information on the source class and
+  the destination class of the reference. Each static linkage error has
+  one of the three types:
+
+  - _Missing class type_ is the error when where the destination class of a
+    class reference does not exist in the linkage classpath. This error
+    happens when a class is removed in a different version of a library,
+    or there is a dependency missed when constructing the linkage classpath.
+    The reference that causes a missing class error is called a _dangling reference_.
+
+  - _Missing method type_ is the error when a method reference has a  static
+    linkage conflict.
+
+  - _Missing field type_ is the error when a field reference has a static
+     linkage conflict.
+
+- **Linkage classpath**: the classpath which a static linkage check
+  runs against. For a Maven project, a linkage classpath consists of the
+  project's class files and the transitive dependencies induced by the direct
+  dependencies in the project's pom.xml file.
+
+- **Class usage graph**: a directed graph that represents the relationship between
```

Comment:
Yes, it's cyclic. Updated.

Added description:

> To avoid unnecessary graph traversal, self-loops (references between the same class) are omitted.